### Data Analysis with Panda
- Video Game Sales 
- Michael Hendricks
- July 21

In [138]:
import pandas as pd
import numpy as np

In [139]:
df = pd.read_csv('data/vgsales.csv')

In [140]:
pub_counts = df[['Publisher']].value_counts()

In [141]:
pub_counts.head()


Publisher                   
Electronic Arts                 1351
Activision                       975
Namco Bandai Games               932
Ubisoft                          921
Konami Digital Entertainment     832
dtype: int64

In [142]:
most_common_publisher = pub_counts[0]

### 1. The Most common Video Game Publisher is Electronic arts

In [143]:
plat_counts = df[["Platform"]].value_counts()

In [144]:
plat_counts.head()

Platform
DS          2163
PS2         2161
PS3         1329
Wii         1325
X360        1265
dtype: int64

In [145]:
most_common_platform = plat_counts[0]

In [146]:
most_common_plat

2163

### 2. The most common platform is DS

In [147]:
genre_counts = df[['Genre']].value_counts()

In [148]:
genre_counts.head()

Genre       
Action          3316
Sports          2346
Misc            1739
Role-Playing    1488
Shooter         1310
dtype: int64

In [149]:
most_common_genre = genre_counts[0]

In [150]:
most_common_genre

3316

### 3. The most common genre is Action

In [151]:
top_twenty_highest_grossing_games = df[['Name','Global_Sales']]

### 4. The top 20 highest grossing games are:

In [152]:
top_twenty_highest_grossing_games.head(20)

Name  Global_Sales
0                                     Wii Sports         82.74
1                              Super Mario Bros.         40.24
2                                 Mario Kart Wii         35.82
3                              Wii Sports Resort         33.00
4                       Pokemon Red/Pokemon Blue         31.37
5                                         Tetris         30.26
6                          New Super Mario Bros.         30.01
7                                       Wii Play         29.02
8                      New Super Mario Bros. Wii         28.62
9                                      Duck Hunt         28.31
10                                    Nintendogs         24.76
11                                 Mario Kart DS         23.42
12                   Pokemon Gold/Pokemon Silver         23.10
13                                       Wii Fit         22.72
14                                  Wii Fit Plus         22.00
15                            Kinect Adventures!         21.82
16                            Grand Theft Auto V         21.40
17                 Grand Theft Auto: San Andreas         20.81
18                             Super Mario World         20.61
19  Brain Age: Train Your Brain in Minutes a Day         20.22

In [153]:
name_NA_only = df[['Name','NA_Sales']]

In [154]:
na_median_sales = name_NA_only.median()

In [155]:
na_median_sales[0]

0.08

In [156]:
df[df['NA_Sales'] == 0.08][['Name','NA_Sales']].sort_values('NA_Sales').tail(10)

Name  NA_Sales
8941                              Galerians      0.08
8939   Disney's Donald Duck: Goin' Quackers      0.08
8936                        Toy Story Racer      0.08
8931                   Kung-Fu: High Impact      0.08
8900                           Cardinal Syn      0.08
8887                         Chessmaster II      0.08
8869            T'ai Fu: Wrath of the Tiger      0.08
8860                 Mirror's Edge Catalyst      0.08
8826                       FIFA Soccer 2002      0.08
11492          Ultimate Shooting Collection      0.08

### 5. For North American video game sales, the median is 0.08 million

In [157]:
standard_dev = df[['NA_Sales']].std()

In [158]:
standard_dev[0]

0.8166830292988796

### 6. The Highest selling video game of all time is 0.8166 standard deviations above the mean

In [159]:
#df[df['Platform'] == 'Wii'][['Name','Platform','Global_Sales']].sort_values('Global_Sales')

In [160]:
wii_games = df[df['Platform'] == 'Wii'][['Name','Platform','Global_Sales']].sort_values('Global_Sales')

In [161]:
#df[df['Platform'] != 'Wii'][['Name','Platform','Global_Sales']].sort_values('Global_Sales')

In [162]:
non_wii_games = df[df['Platform'] != 'Wii'][['Name','Platform','Global_Sales']].sort_values('Global_Sales')

In [163]:
wii_games.mean()

Global_Sales    0.699404
dtype: float64

In [164]:
non_wii_games.mean()

Global_Sales    0.52339
dtype: float64

### 7. Wii games averages 0.699404 million sold per game compared to only 0.52339 million for other platforms

In [165]:
df[['Year']].min()

Year    1980.0
dtype: float64

In [166]:
df[df['Year'] == 1980.0][['Name','Year']].sort_values('Year')

Name    Year
258         Asteroids  1980.0
544   Missile Command  1980.0
1766          Kaboom!  1980.0
1969         Defender  1980.0
2669           Boxing  1980.0
4025       Ice Hockey  1980.0
5366          Freeway  1980.0
6317           Bridge  1980.0
6896         Checkers  1980.0

### 8. The first 9 video games came out in 1980

In [167]:
df[df['Publisher'] == 'Kemco'][['Name','Year']].sort_values('Year').max()

Name    Woody Woodpecker in Crazy Castle 5
Year                                2005.0
dtype: object

### 9. The Last Kemco game was published in 2005

In [168]:
name_global_genre = df[['Name','Global_Sales','Genre']]

### 10. Tetris is the only 'Puzzle' game in the top 20 grossing games

In [169]:
 def tests():   
    def assert_equal(actual,expected):
        assert actual == expected, f"Expected {expected} but got {actual}"

    assert_equal(most_common_publisher, pub_counts[0])
    assert_equal(most_common_platform, plat_counts[0])
    assert_equal(most_common_genre, genre_counts[0])
    assert_equal(top_twenty_highest_grossing_games.iloc[0].Name, 'Wii Sports')
    assert_equal(top_twenty_highest_grossing_games.iloc[19].Name, 'Brain Age: Train Your Brain in Minutes a Day')
    assert_equal(na_median_sales[0], 0.08)
    assert_equal(standard_dev[0], 0.8166830292988796
)
    assert_equal( wii_games.mean()[0] > non_wii_games.mean()[0],True)

    print("Success!!")

In [170]:
tests()

Success!!!
